main.go

```diff
@@ -445,27 +446,50 @@ func main() {
 
 	args := flag.Args()
 
-	arn := ""
+	initialRole := ""
+	roles := []string{}
+
+	// 1. len(args) == 1 => initialRole = args[0]
+	// 2. len(args) == 1, sourceRole != nil => initialRole = *sourceRole, roles = args <= legacy
+	// 3. alias(arn) => initialRole = arn, roles = []
+	// 4. alias(arn, sourceRole) => initialRole = sourceRole, roles = [arn] <= legacy
+	// 5. alias(rolechain) => initialRole = rolechain[0], roles = rolechain[1:]
+	// 5. len(args) > 1 => initialRole = args[0], roles = args[1:]
+
 	if *aliasFlag != "" {
 		alias, ok := oc.Alias[*aliasFlag]
 		if !ok {
 			log.Fatalf("unknown alias: %s", *aliasFlag)
 		}
-		arn = alias.Arn
-		sourceRole = &alias.SourceRole
+		if len(alias.RoleChain) > 0 {
+			initialRole = alias.RoleChain[0]
+			roles = alias.RoleChain[1:]
+		} else {
+			initialRole = alias.Arn
+			if alias.SourceRole != "" {
+				roles = []string{alias.SourceRole}
+			}
+		}
+	} else if *sourceRole != "" {
+		initialRole = *sourceRole
+		roles = args
 	} else {
 		if len(args) > 0 {
-			arn = args[0]
+			initialRole = args[0]
+			roles = args[1:]
 		}
-
 	}
 
-	if arn == "" {
+	if initialRole == "" {
 		log.Fatal("no arn provided in Args[1] or alias")
 	}
 
 	// Check for AWS credentials for role, use them if not expired
-	result, err := credentialsForRole(arn)
+	finalRole := initialRole
+	if len(roles) > 0 {
+		finalRole = roles[len(roles)-1]
+	}
+	result, err := credentialsForRole(finalRole)
```

Comment:
Instead of just checking for the final role, it could potentially walk backwards, eg:

```
oidc2aws r1 r2 r3 r4
```

1. Cached credentials for `r4`? No.
2. Cached credentials for `r3`? Yes.
3. Ok, fetch `r4` using `r3`